redaxo/src/addons/structure/plugins/history/assets/history.js

```diff
@@ -224,6 +224,20 @@
             this.targetFrame = $('#content-history-iframe-2', this.el);
             this.submit = $('[data-history-layer="snap"]', this.el);
             this.cancel = $('[data-history-layer="cancel"]', this.el);
+
+            // fix layout for browsers having flexbox issues (Safari < 11 most notably)
+            // they require a px based flex container to properly calculate childs’ height
+            // unfortunately, this results in removing the responsiveness ¯\_(ツ)_/¯
+            // see: https://github.com/philipwalton/flexbugs#14-flex-containers-with-wrapping-the-container-is-not-sized-to-contain-its-items
+            // use timeout to run late!
+            window.setTimeout(function () {
+                this.wrapper = $('.history-layer-panel-3', this.el);
```

Comment:
diese ziele sollte ins IF sodass dieser selector nur gesucht wird, wenn die element height ensprechend berechnet wurde

Reply:
sollte wrapper hier eine lokale variable sein anstatt an `this.` gebunden?

Reply:
mit obigen CSS könnte sich das JS erledigt haben. @schuer sollte nur einmal gegen testen

Reply:
@staabm Die Instanz wird der Timeout-Funktion mitgegeben, um `wrapper` und `target` bewusst daran binden zu können, weil das gesamte Konstrukt darauf ausgelegt ist, mit potentiell mehreren Historie-Instanzen funktionieren zu können. (Das ist hier zwar nicht der Fall, aber ich wollte es so generisch halten.) `this.el` wird erstmalig selektiert, wenn die Instanz generiert wird, und danach hangelt sich alles weitere an diesem Element runter.

Reply:
Die Idee mit dem padding statt height kommt ja vom Einbinden von Videos.
Insofern wärs vielleicht interessant wie solche Lösungen mit Video auf Safari aussehen. Von da ausgehend kann man weitersehen. Hab glücklicherweise keinen Mac und somit auch kein Safari :D

Reply:
@DavidBruchmann Die Padding-Lösung verwendet man ja, um in einem festen Seitenverhältnis skalieren zu können. Damit hat auch Safari kein Problem. Unser Problem liegt vielmehr an dem sehr speziellen Kontext aus verschachtelten Flexboxen und absolut positionierten Containern, der ausgerechnet diesen sehr nervigen Safari-Bug aufwirft, bei dem Kindelemente die Höhe des Elternelements nicht richtig annehmen. Normalerweise würde man versuchen, den Kontext anzupassen, in unserem Fall ist das dummerweise nicht so einfach möglich, ohne das Ergebnis für gute Browser zu verschlechtern. Deshalb nun das JS-Rumgefixe.

Reply:
Achso, @staabm noch vergessen zu sagen: Das Timeout-Konstrukt ist übrigens nur deshalb drin, weil die Elemente zu dem Zeitpunkt noch nicht gerendert worden sind. Die Höhe kann erst danach gemessen werden, deshalb findet alles innerhalb des Timeouts statt, nicht schon vorher.